flair/models/lemmatization_model.py

```diff
@@ -0,0 +1,617 @@
+import logging
+from pathlib import Path
+from typing import List, Union
+import random
+
+import torch
+import torch.nn as nn
+from torch.utils.data.dataset import Dataset
+import torch.nn.functional as F
+
+import flair.nn
+from flair.data import Dictionary, Sentence
+from flair.datasets import SentenceDataset, DataLoader
+from flair.embeddings import FlairEmbeddings
+from flair.training_utils import Metric, Result
+from flair.file_utils import cached_path
+
+log = logging.getLogger("flair")
+
+start_token: str = '\t'  # Start-of-sentence token
+end_token: str = '\n'  # End-of-sentence token
+
+
+class Lemmatization(flair.nn.Model):
+
+    def __init__(
+            self,
+            hidden_size: int,
+            embeddings: FlairEmbeddings,
+            character_dictionary: Dictionary,
+            n_layers: int,
+            tag_list: list = None,
+            tag_dictionary: Dictionary = None,
+            dropout: float = 0.1,
+            teacher_forcing_ratio: float = 1,
+            longest_word_length: int = 50,
+            contextualized_embedding: bool = False
+    ):
+        """
+        :param hidden_size: number of hidden states in RNN
+        :param embeddings: pre-trained embeddings
+        :param character_dictionary: dictionary containing all the characters in the corpus
+        :param n_layers: number of RNN layers
+        :param tag_list: list of tags planned to be used. Using tag can improve the accuracy of the model,
+        Using tag can improve the accuracy of the model, but the model trained with tag does not work well with data without tag.
+        :param tag_dictionary: dictionary of tags you need to use.(The all possible values of the tag in tag_list)
+        :param dropout: dictionary contains all the tag values in the tag list
+        :param teacher_forcing_ratio: the probability of using teacher_forcing. If it is greater than or equal to 1,
+        always use teacher forcing. If less than or equal to 0, never use.
+        :param longest_word_length: record the maximum length of words and lemma to use in prediction.
+        Based on experience, the default value is set at 50.
+        :param contextualized_embedding: Whether to use context-sensitive word embedding. Using context-sensitive word
+        embedding can improve the accuracy of the model, but it takes more time for training and prediction.
+        """
+
+        super(Lemmatization, self).__init__()
+        self.hidden_size = hidden_size
+        self.embeddings = embeddings.to(flair.device)
```

Comment:
I think the `.to(flair.device)` is unnecessary here. Embeddings are torch modules so they will be moved together with the Lemmatization model if you send it to a device.

Reply:
Exactly! I will fix it.